src/lib/builtins/array.rs

```diff
@@ -470,6 +470,40 @@ pub fn find_index(this: &Value, args: &[Value], interpreter: &mut Interpreter) -
     Ok(Gc::new(ValueData::Number(f64::from(-1))))
 }
 
+/// Array.prototype.fill ( value[, start[, end]] )
+///
+/// The method fills (modifies) all the elements of an array from start index (default 0)
+/// to an end index (default array length) with a static value. It returns the modified array
+/// <https://tc39.es/ecma262/#sec-array.prototype.fill>
+pub fn fill(this: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+    let len: i32 = from_value(this.get_field_slice("length")).expect("Could not get argument");
+    let default_value = Gc::new(ValueData::Undefined);
```

Comment:
Use the [`undefined()`](https://github.com/jasonwilliams/boa/blob/master/src/lib/builtins/value.rs#L23) function here.